src/raster/array.jl

```diff
@@ -1,89 +1,106 @@
+import DiskArrays: eachchunk, GridChunks
+import DiskArrays
 const AllowedXY = Union{Integer,Colon,AbstractRange}
 const AllowedBand = Union{Integer,Colon,AbstractArray}
 
 
+# Define a RasterDataset type
```

Comment:
I'm going by the reactions in https://github.com/meggart/DiskArrays.jl/issues/2#issue-549445915 to guess that there is general acceptance of DiskArrays as an interface: in that case, it will be good to elaborate on what a RasterDataset is -- its correspondence to the GDAL Data Model, and how it relates to e.g. Raster[Band] versus as a Dataset, so that the discussion and conceptual models are contained in the code without being tied to github.